pennylane/devices/qubit/simulate.py

```diff
@@ -65,6 +65,40 @@
 }
 
 
+class TreeTraversalStack:
+    """This class is used to record various data used during the
+    depth-first tree-traversal procedure for simulating dynamic circuits."""
+
+    counts: dict
+    probs: dict
+    results_0: Sequence
+    results_1: Sequence
+    states: Sequence
+
+    def __init__(self, n):
+        self.counts = [None] * n
+        self.probs = [None] * n
```

Comment:
> The type hint for ...

I started with list-valued dicts but ended up with lists of dicts. I'll change it.

> Also, what is n?

It is the max stack depth, which is `n_mcms` + 1. I thought it would be misleading calling it `n_mcms`, so I just used a generic name. Maybe `max_depth` would be better then. What do you think?

Reply:
Yeah `max_depth` sounds great :)